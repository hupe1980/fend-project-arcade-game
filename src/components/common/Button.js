import styled from 'styled-components';

const Button = styled.button`
  border: 2px solid blue;
  background-color: white;
  color: blue;
  padding: 14px 28px;
  font-size: 16px;
  cursor: pointer;

  &:hover {
    background-color: blue;
    color: white;
  }
`;

export default Button;
