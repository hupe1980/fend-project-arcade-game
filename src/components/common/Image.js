import styled from 'styled-components';

const Image = styled.img`
  width: 34px;
  height: 57px;
  margin: -10px 0 0 0;
`;

export default Image;
