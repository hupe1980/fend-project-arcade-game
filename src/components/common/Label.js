import styled from 'styled-components';

const Label = styled.span``;

export default Label;
