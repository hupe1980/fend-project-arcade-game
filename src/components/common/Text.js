import styled from 'styled-components';

const Text = styled.p`
  color: blue;
`;

export default Text;
