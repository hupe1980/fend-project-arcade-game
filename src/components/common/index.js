export { default as Button } from './Button';
export { default as Heading } from './Heading';
export { default as LoadingIndicator } from './LoadingIndicator';
export { default as Label } from './Label';
export { default as Image } from './Image';
export { default as Text } from './Text';
export { default as withAnimationFrame } from './withAnimationFrame';
