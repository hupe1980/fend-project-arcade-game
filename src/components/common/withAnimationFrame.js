import React, { Component } from 'react';

export default function withAnimationFrame(WrappedComponent) {
  return class AnimatedComponent extends Component {
    constructor(props) {
      super(props);

      this._frame = null;

      this.requestAnimationFrame = this.requestAnimationFrame.bind(this);
      this._loop = this._loop.bind(this);
      this.cancelAnimationFrame = this.cancelAnimationFrame.bind(this);
    }

    componentDidMount() {
      this.requestAnimationFrame();
    }

    componentWillUnmount() {
      this.cancelAnimationFrame();
    }

    requestAnimationFrame() {
      this._frame = window.requestAnimationFrame(this._loop);
    }

    _loop(time) {
      const target = this.refs.target;

      target.onAnimationFrame(time);

      this.requestAnimationFrame();
    }

    cancelAnimationFrame() {
      window.cancelAnimationFrame(this._frame);
    }

    render() {
      return (
        <WrappedComponent
          ref="target"
          requestAnimationFrame={this.requestAnimationFrame}
          cancelAnimationFrame={this.cancelAnimationFrame}
          {...this.props}
        />
      );
    }
  };
}
