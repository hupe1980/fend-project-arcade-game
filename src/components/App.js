import React, { Component } from 'react';
import styled from 'styled-components';

import Game from './Game';
import AppBar from './AppBar';
import Footer from './Footer';
import { LoadingIndicator } from './common';
import { preloadImages, GAME_WIDTH, GAME_HEIGHT } from '../game';

const Container = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 700px;
  min-width: ${GAME_WIDTH}px;
  margin: auto;
`;

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { loading: true };
  }

  componentDidMount() {
    preloadImages(() => this.setState({ loading: false }));
  }

  render() {
    if (this.state.loading) return <LoadingIndicator />;

    return (
      <Container>
        <AppBar />
        <Game width={GAME_WIDTH} height={GAME_HEIGHT} />
        <Footer />
      </Container>
    );
  }
}
