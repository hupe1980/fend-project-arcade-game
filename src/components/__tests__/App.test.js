import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../App';
import { LoadingIndicator } from '../common';

describe('App', () => {
  it('renders correctly (state = { loading: true })', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<LoadingIndicator />)).toEqual(true);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders App (state = { loading: false })', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ loading: false });
    expect(wrapper.contains(<LoadingIndicator />)).toEqual(false);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
