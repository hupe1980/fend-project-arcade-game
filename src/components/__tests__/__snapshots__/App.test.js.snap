// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App (state = { loading: false }) 1`] = `
<styled.main>
  <Connect(Component) />
  <Connect(Component)
    height={606}
    width={505}
  />
  <Footer />
</styled.main>
`;

exports[`App renders correctly (state = { loading: true }) 1`] = `<LoadingIndicator />`;
